### PR TITLE
Add body to http status error

### PR DESCRIPTION
**Pull request by imclerran**
Fix for issue #207. One change from the fix described in the issue:

Rather than have `body: List U8`, body is an `ErrorBody` instead, which has a custom Inspect implementation. This enables two things: 
- First, it truncates the body to 50 bytes to keep the Inspect output short. 
- Second, it converts the body to Str so that inspect shows something usable to the user, rather than a list of bytes.
    - If conversion to Str fails, it simply inspects as as the following str: "Invalid UTF-8 data"
    - This is with the understanding that when inspecting a BadStatus error, both Http code and body will be present, and the code will still give usable info to the user, even if body does not convert to string. 
    - With this said, I am looking forward to feedback in review on this handling of Inspect.
    
  
Since the type of a bad status error is now: `BadStatus { code: U16, body: ErrorBody }`, where ErrorBody is opaque, the Http module now exposes a new function: `errorBodyToBytes`.

Last thing of note: errorToString currently converts the entire body to string (if possible. If not, it simply includes the code), while Inspect only takes the first 50 bytes.

### CODE REVIEW

File: platform/InternalHttp.roc [hunk 1]
```diff
@@ -37,6 +37,25 @@ Error : [
...
+            toInspector: errorBodyToInspector,
+        },
+    ]
+
+errorBodyToInspector : ErrorBody -> _
+errorBodyToInspector = \@ErrorBody val ->
+    fmt <- Inspect.custom
+    when val |> List.takeFirst 50 |> Str.fromUtf8 is
+        Ok str -> Inspect.apply (Inspect.str str) fmt
+        Err _ -> Inspect.apply (Inspect.str "Invalid UTF-8 data") fmt
```
Comment (lukewilliamboswell) [inline platform/InternalHttp.roc:55]: I think this is a great idea.

**Review by lukewilliamboswell — APPROVED** [review]
Looks good to me. 😄

File: platform/InternalHttp.roc (hunk 1 above)
Comment (imclerran) [reply platform/InternalHttp.roc:55]: Great!

**Comment by lukewilliamboswell** [comment]
> New requirement to sign all commits?

No - it's the same with all of the roc repositories. 😄

**Comment by imclerran** [comment]
I haven't signed commits in the past 🤔. Do I need to create a new PR with only signed commits?

**Comment by lukewilliamboswell** [comment]
I don't think you need to do that. I think you could try something like `git rebase --exec 'git commit --amend --no-edit -n -S' -i HEAD~3`

**Comment by imclerran** [comment]
All commits signed, should be good to go! 😁

**Review by lukewilliamboswell — APPROVED** [review]

**Pull request merged**